File: LeetCode/LeetCode/HackerRank/Problem Solving/DiagonalDifference.cs
using System;
using System.Collections.Generic;

namespace LeetCode.HackerRank.Problem_Solving;

public class DiagonalDifference
{
    public static int diagonalDifference(List<List<int>> arr)
    {
        var length = arr.Count;
        var diagonalSumRight = 0;
        var diagonalSumLeft = 0;

        for (var i = 0; i < length; i++)
        {
            var j = length - 1 - i;
            diagonalSumRight += arr[i][i];
            diagonalSumLeft += arr[i][j];
        }

        var difference = Math.Abs(diagonalSumRight - diagonalSumLeft);

        return difference;
    }
}

File: LearningTypeScript/src/HackerRank/ProblemSolving/Arrays/DiagonalDifference.ts
export function diagonalDifference(arr: number[][]): number {

    let RightSum: number = 0;
    let LeftSum: number = 0;
    let lastIndex: number = arr.length - 1;

    for(let i = 0; i <= lastIndex; i++) {

        let j = lastIndex - i;

        RightSum += arr[i][i];
        LeftSum += arr[i][j];
    }

    return Math.abs(RightSum - LeftSum);
}